example/nvtx_test/test.cpp

```diff
@@ -0,0 +1,162 @@
+  License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+#include <iostream>
+#include <iomanip> 
+#include <string>    
+#include <vector>     
+#include "mpi.h"
+#include "omp.h"
+#include "phiprof.hpp"
+
+using namespace std;
+
+double compute(double seconds){
+   const double overhead=0.000002; //estimated
+   double a,b,c,d;
+   int  i,iterations=1000;
+   double t1,t2;
+
+   a=1.00004;
+   b=1e-10;
+   c=1.00011;
+   d=0;
+
+   t2=t1=MPI_Wtime();
+   while(t2-t1<seconds-overhead){
+      #pragma acc parallel loop
+      for(i=0;i<iterations;i++){
+         d+=a*b+c;
+      }
+      t2=MPI_Wtime();
+   }
+   return d;
+}
+
+
+int main(int argc,char **argv){
+
+   int rank;
+   MPI_Init(&argc,&argv);
+   MPI_Comm_rank(MPI_COMM_WORLD,&rank);
+   //const int nIterations=1000000;
+   const int nIterations=10;
+   if(rank==0)
+      cout << "Measuring performance of start-stop calls" <<endl;
+   
+   phiprof::initialize();
+   
+   phiprof::start("Benchmarking phiprof"); 
+
+   phiprof::start("Initalized timers using ID");
+
+   if(rank==0)
+      cout << "  1/3" <<endl;
+   int id_a=phiprof::initializeTimer("a","A with ID");
+   for(int i=0;i<nIterations;i++){
+      phiprof::start(id_a);
+      phiprof::stop(id_a);
+   }
+   phiprof::stop("Initalized timers using ID",nIterations,"start-stop");
+
+   if(rank==0)
+      cout << "  2/3" <<endl;
+
+   phiprof::start("Re-initialized timers using ID");
+   for(int i=0;i<nIterations;i++){
+      id_a=phiprof::initializeTimer("a","A with ID");
+      phiprof::start(id_a);
+      phiprof::stop(id_a);
+   }
+   phiprof::stop("Re-initialized timers using ID",nIterations,"start-stop");
+
+   if(rank==0)
+      cout << "  3/3" <<endl;
+   phiprof::start("Timers using labels");
+   for(int i=0;i<nIterations;i++){
+      phiprof::start("a");
+      phiprof::stop("a");
+   }
+   phiprof::stop("Timers using labels", nIterations*2, "start-stop");
+   phiprof::stop("Benchmarking phiprof"); 
+
+   MPI_Barrier(MPI_COMM_WORLD);
+ 
+   if(rank==0)
+      cout << "Measuring accuracy" <<endl;
+
+
+   phiprof::start("Test accuracy");
+
+   if(rank==0)
+      cout << "  1/3" <<endl;
+   phiprof::start("100x0.01s computations"); 
+   for(int i=0;i<100;i++){
+      phiprof::start("compute");
+      compute(0.01);
+      phiprof::stop("compute");
+   }
+   phiprof::stop("100x0.01s computations");
+
+   if(rank==0)
+      cout << "  2/3" <<endl;
+   MPI_Barrier(MPI_COMM_WORLD);
+
+   phiprof::start("100 x 0.01 (threadId + 1)s, ID");
+   int id = phiprof::initializeTimer("compute");
+#pragma omp parallel
```

Comment:
should this test not be with openacc? openmp is not even enabled in makefile

Reply:
I thought it would have one openacc parallel loop and also some openmp stuff for completeness.

Reply:
Check!